test/e2e/memcached_test.go

```diff
@@ -120,30 +118,21 @@ func TestMemcached(t *testing.T) {
 			// memcached-operator's go.mod, which allows go to recognize
 			// the local SDK repo as a module.
 			sdkModPath := filepath.Join(filepath.FromSlash(replace.repo), "go.mod")
-			err = ioutil.WriteFile(sdkModPath, []byte("module "+sdkRepo), fileutil.DefaultFileMode)
-			if err != nil {
-				t.Fatalf("Failed to write main repo go.mod file: %v", err)
-			}
-			defer func() {
-				if err = os.RemoveAll(sdkModPath); err != nil {
-					t.Fatalf("Failed to remove %s: %v", sdkModPath, err)
+			if _, err = os.Stat(sdkModPath); err != nil && os.IsNotExist(err) {
```

Comment:
What's supposed to happen if `err != nil && os.IsNotExist(err) == false`, i.e some other error (permissions, filesystem)?
Looking down it seems like `insertGoModReplace()` will fail when it tries to read the file so I guess it's fine.

Reply:
Yep I figured I'd let `ioutil.ReadFile()` handle that error.